File: Data Structures & Algorithms/find-duplicate-integer/submission-1.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        n = len(nums)
        i = 0
        while i<n:
            x = nums[i]
            if (i+1) == x:
                i += 1
                continue
            if nums[x-1] == x:
                return x
            nums[i], nums[x-1] = nums[x-1], nums[i]
        return 0

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        vector<int> ret;
        priority_queue<pair<int, int>> q;
        for (int i=0; i<nums.size(); i++) {
            q.push({nums[i], i});
            if (i>=k-1) {
                while (q.top().second<i-k+1) {
                    q.pop();
                }
                ret.push_back(q.top().first);
            }
        }
        return ret;
    }
};
